ruby/private/sdk.bzl

```diff
@@ -1,15 +1,20 @@
 load(
-    ":host_runtime.bzl",
-    _ruby_host_runtime = "ruby_host_runtime",
+    "@bazelruby_ruby_rules//ruby/private/toolchains:ruby_runtime.bzl",
+    _ruby_runtime = "ruby_runtime",
 )
 
-def _register_host_runtime():
-    _ruby_host_runtime(name = "org_ruby_lang_ruby_host")
+def ruby_register_toolchains(version = "host"):
+    """Registers ruby toolchains in the WORKSPACE file."""
+
+    supported_versions = ["host", "2.6.3", "2.6.5"]
+    if version in supported_versions:
+        _ruby_runtime(
+            name = "org_ruby_lang_ruby_toolchain",
```

Comment:
This makes cross build impossible.
It is still better to call this runtime `org_ruby_lang_ruby_host` or `ruby_runtime_host` or something.

Reply:
@yugui I think I am getting confused between two concepts. 
1. Cross compiling an interpreter
2. Setting up an interpreter in the toolchain

As I understand it:
I can't think of a good reason to have a ruby interpreter in the toolchain that is compiled for another architecture. This would make running things like `bundler` `rspec` and `rubocop` impossible because the toolchains ruby would not be able to run with something like `bazel run :test`

As I see in your latest changes to `ruby_binary` you see the option of having `host` as being more like `dont_include_interpreter`, i.e.  1ruby_binary(...)` would include the ruby interpreter if the toolchains `is_host` is `False`.

I think we should separate these concepts, and maybe introduce a rule `ruby_runtime(version="2.5", arch="x64")` or something that we can put into a binary to bundle them together, e.g.

```
ruby_runtime(name="ruby_2_6_3", version="2.6.3")
ruby_binary(
  ...
  runtime=":ruby_2_6_3"
)
```

For the majority of our use-cases (mostly due to the complexity of cross compilation + native gem building) we don't want the interpreter bundled with our binaries.

Reply:
Thank you for your answer.
To be brief, now I agree to your naming `org_ruby_lang_ruby_toolchain`.

> I can't think of a good reason to have a ruby interpreter in the toolchain that is compiled for another architecture. 
* IIUC, Bazel can distinguish the interpreter for the build environment and the foreign environments as far as their target platforms are correctly configured.
* Ruby allows us to cross-compile extensions with the headers and `rbconfig.rb` for the target platform

It is certainly a little hard to setup up a cross-compile environment, personally I'd like to keep the door open.

>  you see the option of having host as being more like `dont_include_interpreter`

Right. It is better to rename the option.

> ruby_runtime(version="2.5", arch="x64") 

Good point. It is definitely a better API.
Also I started think we needed to distinguish toolchain and runtime.
* toolchain is used to fetch gems; and interpret helper ruby scripts, and provides a primary runtime
* runtime is used to build C extensions with it and optionally included into the `ruby_binary`.

Considering that, it should be safe to call the toolchain just `org_ruby_lang_ruby_toolchain`.

Reply:
Awesome.